openqa-investigate

```diff
@@ -1,391 +1,178 @@
+            job_result = job_data['job'].get('result')
+            info = f"* Name: {job_name}\n* Result: {job_result}\n* Reason: {reason}\n\nIt might be a product bug, an outdated needle, test code needing adaptation or a test infrastructure related problem.\nAdding a [bugref](http://open.qa/docs/#_bug_references) that can be [carried over](http://open.qa/docs/#carry-over) will prevent these mails for this issue. If the carry-over is not sufficient, you may want to create a ticket with [auto-review-regex](https://github.com/os-autoinst/scripts/blob/master/README.md#auto-review---automatically-detect-known-issues-in-openqa-jobs-label-openqa-jobs-with-ticket-references-and-optionally-retrigger)."
+            email = f"[{job_name}]({testurl})\n{header}\n{info}\n\n{excerpt}"
+            subject = f"Unreviewed issue (Group {group_id} {group_name})"
+            if not dry_run:
+                send_email(group_mailto, email, from_email, subject)
+
+def send_email(mailto, email, from_email, subject):
+    # This function will just print the email to the console in this example
+    print(f"To: {mailto}\nFrom: {from_email}\nSubject: {subject}\n\n{email}")
+
+def get_log_file(job_id):
+    # Placeholder for actual log file retrieval logic
+    return f"/path/to/logs/{job_id}.log"
+
+def get_log_excerpt(log_file):
+    try:
+        with open(log_file, 'r') as f:
+            lines = f.readlines()
+            excerpt = ""
+            # Simulated logic for getting log excerpts
+            for line in lines[-15:]:
+                excerpt += line
+            return excerpt
+    except FileNotFoundError:
+        return "(No log excerpt found)"
+
+def query_dependency_data_or_postpone(job_id, job_data):
+    response = requests.get(f"{host_url}/tests/{job_id}/dependencies_ajax")
+    if response.status_code == 404:
+        warn(f"Job {job_id} dependencies not found, postponing")
+        return 142
+    response.raise_for_status()
+    return response.json()
+
+def sync_via_investigation_comment(job_id, first_cluster_job_id):
+    comments = requests.get(f"{host_url}/jobs/{job_id}/comments").json()
+    comment_id = next((comment['id'] for comment in comments if comment['text'].startswith("Starting investigation for job")), None)
+    first_comment_id = next((comment['id'] for comment in comments if comment['text'].startswith("Starting investigation for job")), None)
+    if comment_id != first_comment_id:
+        warn(f"Found existing comment id {first_comment_id}, assuming cluster jobs are already investigated and skipping")
+        requests.delete(f"{host_url}/jobs/{first_cluster_job_id}/comments/{comment_id}")
+        return 255
+    return 0
+
+def main():
+    parser = argparse.ArgumentParser(description="Trigger investigation jobs")
+    parser.add_argument("job_id", type=int, help="Job ID to investigate")
+    parser.add_argument("additional_params", nargs="*", help="Additional parameters for job investigation")
```

Comment:
maybe the args can go outside the main and passed in the main to improve readibility?